src/BoundaryConditions/fill_halo_regions.jl

```diff
@@ -22,29 +22,24 @@ conditions, possibly recursing into `fields` if it is a nested tuple-of-tuples.
 fill_halo_regions!(c::OffsetArray, ::Nothing, args...; kwargs...) = nothing
 
 for dir in (:west, :east, :south, :north, :bottom, :top)
-    extract_bc = Symbol(:extract_, dir, :_bc)
+    extract_side_bc = Symbol(:extract_, dir, :_bc)
     @eval begin
-        @inline $extract_bc(bc) = bc.$dir
-        @inline $extract_bc(bc::Tuple) = map($extract_bc, bc)
+        @inline $extract_side_bc(bc) = bc.$dir
+        @inline $extract_side_bc(bc::Tuple) = map($extract_side_bc, bc)
     end
 end
 
-# For inhomogeneous BC we extract the _last_ one 
-# example 
-# `bc.west <: DCBC`
-# `bc.east <: PBC`
-# `extract_west_or_east_bc(bc) == bc.west`
-# NOTE that `isless` follows order of execution, 
-# so `max(bcs...)` returns the last BC to execute
-
-  extract_west_or_east_bc(bc) = max(bc.west,   bc.east)
-extract_south_or_north_bc(bc) = max(bc.south,  bc.north)
- extract_bottom_or_top_bc(bc) = max(bc.bottom, bc.top)
-
-  extract_west_or_east_bc(bc::Tuple) =   map(extract_west_or_east_bc, bc)
-extract_south_or_north_bc(bc::Tuple) = map(extract_south_or_north_bc, bc)
- extract_bottom_or_top_bc(bc::Tuple) =  map(extract_bottom_or_top_bc, bc)
- 
+@inline extract_bc(bc, ::Val{:west})   = tuple(extract_west_bc(bc))
+@inline extract_bc(bc, ::Val{:east})   = tuple(extract_east_bc(bc))
+@inline extract_bc(bc, ::Val{:south})  = tuple(extract_south_bc(bc))
+@inline extract_bc(bc, ::Val{:north})  = tuple(extract_north_bc(bc))
+@inline extract_bc(bc, ::Val{:bottom}) = tuple(extract_bottom_bc(bc))
+@inline extract_bc(bc, ::Val{:top})    = tuple(extract_top_bc(bc))
+
+@inline extract_bc(bc, ::Val{:west_and_east})   = (extract_west_bc(bc), extract_east_bc(bc))
+@inline extract_bc(bc, ::Val{:south_and_north}) = (extract_south_bc(bc), extract_north_bc(bc))
+@inline extract_bc(bc, ::Val{:bottom_and_top})  = (extract_bottom_bc(bc), extract_top_bc(bc))
```

Comment:
What's all this for?

Reply:
Oh okay, so this will extract a "tuple of west bcs" and a "tuple of east bcs".

Can you write that in a comment?